Automattic/jetpack
# Refactor checkout modal

Fixes #32237
Fixes https://github.com/Automattic/wp-calypso/issues/71048
Fixes 153-gh-automattic/gold

## Proposed changes:

* A refactor of the checkout modal for memberships which removes the Thickbox and jQuery dependencies.

### Other information:


## Jetpack product discussion



## Does this pull request change what data or activity we track or use?



## Testing instructions:






* Optionally, you can apply D131322-code to your sandbox. This isn't required for this test, but if you choose to do so, make sure you download the patch in the next step on top of this patch (and not to trunk).
* Download this patch to your sandbox.
* Use a test site with payment plans and multiple membership block types.
* Navigate to a site with a payment button / paywalled content.
* Click the button / enter your email and click subscribe to bring up the checkout modal.
* Observe that the checkout modal functions as expected.
* Confirm there is no jquery or thickbox dependency loaded in the browser for these pages.
* Confirm the checkout works for these pages.
